### ISSUE TITLE
Issue : Installing directly from source

### ISSUE THREAD

**Issue by subhas2016** [spyder-ide/spyder#3247]
## Description of your problem
I would like Installing directly from source, I have followed below steps,
 Please help me if in case any wrong steps being followed here.

**What steps will reproduce the problem?**
https://bitbucket.org/spyder-ide/spyderlib/downloads

spyder-3.0.0b3.tar.gz and instatted : Python35-32

spyder-3.0.0b3>python setup.py install

up.py install
WARNING: unable to build documentation because Sphinx is not installed
running install
running build
running build_py
running build_scripts
running install_lib
running install_scripts
running install_data
running install_egg_info
Removing X\Users\XXXX\AppData\Local\Programs\Python\Python35-32\Lib\site-p
ackages\spyder-3.0.0b3-py3.5.egg-info
Writing X\Users\XXXX\AppData\Local\Programs\Python\Python35-32\Lib\site-pa
ckages\spyder-3.0.0b3-py3.5.egg-info

Could you please let me know how to luch the application 

Please note , I have instated 

Executing Spyder from source checkout
Revision None, Branch: None
01. Patched sys.path with D:\Users\703024177\My Data\GE Internal\DIGITAL\PHYTHON
\spyder-3.0.0b3
02. PyQt5 is detected, selecting
Traceback (most recent call last):
  File "bootstrap.py", line 140, in <module>
    versions = get_versions(reporev=False)
  File "D:\Users\703024177\My Data\GE Internal\DIGITAL\PHYTHON\spyder-3.0.0b3\sp
yderlib\__init__.py", line 67, in get_versions
    import qtpy
ImportError: No module named 'qtpy'
**What is the expected output? What do you see instead?**

**Please provide any additional information below**

## Versions and main components

* Spyder Version:spyder-3.0.0b3
* Python Version: Python35-32
* Operating system: Windows

## Dependencies

Please go to the menu entry `Help > Optional Dependencies` (or
`Help > Dependencies`), press the button `Copy to clipboard`
and paste the contents below:

**Comment by ccordoba12** [comment]
Installations from source are discouraged now, unless you're a power user and you know exactly what you're doing.

You have two options here:

1. Track and install all our dependencies (as they are listed in our README.md file)
2. Use pip like this

        pip install --pre -U spyder

    That will install all our dependencies for you.

---

Finally, let me say that we don't plan to do anything else to fix that situation because we prefer our users to use `pip` (or `conda`) to install Spyder.

**Comment by subhas2016** [comment]
I have ran this (pip)command, got the below response, 

Requirement already up-to-date: spyder in \programs\python\python35-32\lib\site-packages.

Please can you suggest how to launch the application? There is no proper documents I found to setup the application. Please suggest on this.

**Comment by Nodd** [comment]
Just run `spyder` in the console.

**Comment by subhas2016** [comment]
Hi

Thanks for your help, I have tried  I got this error ..

ImportError: cannot import name 'PLUGIN_PATH'

Please help

**Comment by subhas2016** [comment]
\spyder-3.0.0b3>spyder
Traceback (most recent call last):
  File "Scripts\spy
der", line 2, in <module>
    from spyderlib import start_app
  File "\lib\site-pa
ckages\spyderlib\start_app.py", line 12, in <module>
    from spyderlib.baseconfig import get_conf_path, running_in_mac_app
  File "\Python\Python35-32\lib\site-pa
ckages\spyderlib\baseconfig.py", line 188, in <module>
    from spyderlib.otherplugins import PLUGIN_PATH
ImportError: cannot import name 'PLUGIN_PATH'

**Comment by subhas2016** [comment]
Hi

Thanks for your help, I have tried I got this error ..

ImportError: cannot import name 'PLUGIN_PATH'

Please help

**Comment by Nodd** [comment]
It looks like your install is all mixed up, the traceback of the error indicates that it's not spyder 3.0.0b3 which is run, but an older one (spyder2.X I suppose). Maybe try to uninstall and reinstall spyder ?

**Comment by subhas2016** [comment]
I have uninstalled completely. then started freshly with below steps:--->

1. Installed  Python(Python 3.5.1)
2. installed  PyQt5-5.6-gpl-Py3.5-Qt5.6.0.
3 The Installed spyder-3.0.0b3 .
4. Run below command pip install --pre -U spyder (Requirement already up-to-date: spyder in local\programs\python\python35-32\lib\site-packages )
5. Then in the console run "spyder " 
**RuntimeError: Failed to import qtpy.**
Please check Spyder installation requirements: qtpy and either PyQt5 5.2+ or PyQt4 4.6+ are required to run Spyder.

Please help me to fix this issue.. many thanks for your help and support on this.

**Comment by Nodd** [comment]
in your step 2, don't install spyder. The pip command is the one that will install spyder.

**Comment by subhas2016** [comment]
Appreciate your help and support on this.
I have followed the same, while in the console run "spyder " getting error python has stopped working, attached print screen for your reference and please let me know in case any more information required .

Please help on this, we almost closer to this matter .

![spyder](https://cloud.githubusercontent.com/assets/20093811/16328668/07261718-39fa-11e6-8cb3-7a0c43f6f3ca.JPG)

**Comment by ccordoba12** [comment]
If you are new to Scientific Python, please follow the instructions present in our README. That will save you a lot of time.

In essence, please:

1. Remove the Python version you have.
2. Download and install a Scientific Python distribution. If you want to use PyQt5, then the only one that supports it at the moment is WinPython: http://winpython.github.io

----

I don't want to be harsh, but we really don't have time to help every user as you're asking for help here (we have about 500.000 users!!). So, as I said before, if you don't understand things too well, please do you and us a favor and follow the easiest path (as I described it above). Else, I'm sorry to say it but you're on your own.

**Comment by ccordoba12** [comment]
Our README is here:

https://github.com/spyder-ide/spyder/blob/master/README.md

**Comment by KrunalDatascience** [comment]
@ccordoba12 : Hi, I am facing the issue as @subhas2016 was facing (ImportError: cannot import name 'PLUGIN_PATH'). After reading your comments I have tried downloading Spyder 3.0.0b using following command mentioned in the readme file:
     conda update qt pyqt
     conda install -c spyder-ide spyder==3.0.0b4

When I executed the second command, conda listed down few packages that are also to be downloaded and installed. When I gave the permission by entering "Y", it downloaded the first package from the list "lazy-object-pr ". And then it started asking me for  - https user name and password. It seems this is because of my PC is behind the corporate firewall. Request you to provide any solution. OR list of URLs that xonda hits to install spyder. I can ask my IT team to whitelist those URLs. Thanks!!!
